### PR TITLE
Fix issue with screen readers not able to see navigation

### PR DESCRIPTION
**Pull request by Guntrisoft**
- Desktop screen reader users would not have been able to
  access the site navigation as it had a aria-hidden=true
  on it
- Rewritten navigation JS as es6 module
- Module now makes nav aria-hidden only on mobile, until
  they click on the menu button

### CODE REVIEW

File: app/assets/javascripts/components/Navigation.es6 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    setNavigationHiddenStatus() {
+      this.$target.attr('aria-hidden', this.getPageSize() === this.mobileSize ? true : false);
+    }
+
+    bindEvents() {
+      $(window).on('resize', this.debounce(this.setNavigationHiddenStatus.bind(this), 50));
+
+      this.$trigger.click(this.handleClick.bind(this));
```
Comment (benbarnett) [inline app/assets/javascripts/components/Navigation.es6]: Can we use the `on` style here too for consistency?

File: app/assets/javascripts/components/Navigation.es6 [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    constructor($component) {
+      super($component);
+      this.init();
+    }
+
+    init() {
+      this.$trigger = $('.js-nav-toggler');
+      this.$target = $('.js-nav');
+      this.$nav = $('.l-page-nav');
+      this.mobileSize = '"mobile"';
```
Comment (benbarnett) [inline app/assets/javascripts/components/Navigation.es6]: 🤢

File: app/assets/javascripts/components/Navigation.es6 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    getPageSize() {
+      const el = window.getComputedStyle(this.$nav[0], ':after');
+
+      return el ? el.getPropertyValue('content') : this.mobileSize;
+    }
+
+    setNavigationHiddenStatus() {
+      this.$target.attr('aria-hidden', this.getPageSize() === this.mobileSize ? true : false);
```
Comment (benbarnett) [inline app/assets/javascripts/components/Navigation.es6]: You could use `this.getPageSize().indexOf(this.mobileSize)` here to get around that double quote shenanigan

File: app/assets/javascripts/components/Navigation.es6 (hunk 2 above)
Comment (Guntrisoft) [inline app/assets/javascripts/components/Navigation.es6]: Yup, agree it's nasty!

File: app/assets/javascripts/components/Navigation.es6 (hunk 3 above)
Comment (Guntrisoft) [inline app/assets/javascripts/components/Navigation.es6]: Done

File: app/assets/javascripts/components/Navigation.es6 (hunk 1 above)
Comment (Guntrisoft) [inline app/assets/javascripts/components/Navigation.es6]: Done

**Comment by Guntrisoft** [comment]
@benbarnett added aria tests

File: app/assets/javascripts/components/BaseComponent.es6 [hunk 4]
```diff
@@ -11,6 +11,26 @@
...
+            timeout = null;
+            if (!immediate) func.apply(context, args);
+          },
+          callNow = immediate && !timeout;
+
+        clearTimeout(timeout);
+
+        timeout = setTimeout(later, wait);
+
+        if (callNow) func.apply(context, args);
```
Comment (benbarnett) [inline app/assets/javascripts/components/BaseComponent.es6]: Actually thought this wouldn't get through the linting. What do you reckon to enabling the [curly](http://eslint.org/docs/rules/curly) rule? Any opinions either way?

Comment (Guntrisoft) [inline app/assets/javascripts/components/BaseComponent.es6]: yup, that seems fair enough - will sort

Comment (Guntrisoft) [inline app/assets/javascripts/components/BaseComponent.es6]: Done

**Comment by benbarnett** [comment]
Cool, but we should separate out the linting config change into its own commit, apply that first, and then add the code change.

We should also explain why we're making the change, even if it's just to explain that we're linting for a standard style the team has already adopted.

**Comment by Guntrisoft** [comment]
@benbarnett agreed - have rebased to split it out

**Pull request merged**